boot/boot_test.go

```diff
@@ -902,3 +902,457 @@ func (s *bootenv20Suite) TestMarkBootSuccessful20BaseUpdate(c *C) {
 	c.Assert(m3.TryBase, Equals, "")
 	c.Assert(m3.BaseStatus, Equals, "")
 }
+
+// TestHappyMarkBootSuccessfulKernelSetBootVarsPanics emulates a reboot during
+// SetBootVars, for the kernel snap when we commit the boot state during
+// MarkBootSuccessful
+func (s *bootenv20Suite) TestHappyMarkBootSuccessfulKernelUpgradeSetBootVarsPanics(c *C) {
```

Comment:
probably the name of the test should refer to RebootBeforSetBootVars instead of Panics which is just a mechanical point